pedpy/io/trajectory_loader.py

```diff
@@ -651,3 +655,389 @@ def _load_trajectory_data_from_viswalk(
+            will be ignored.
+
+    Returns:
+        The trajectory data as :class:`DataFrame`, the coordinates are in meter (m).
+    """
+
+    VADERE_COMMENT = "#"  # Comment identifier in Vadere trajectory files
+    VADERE_KEY_ID = "pedestrianId"
+    VADERE_KEY_TIME = "simTime"
+    VADERE_KEY_X = "startX"
+    VADERE_KEY_Y = "startY"
+    columns_to_keep = [
+        VADERE_KEY_ID,
+        VADERE_KEY_TIME,
+        VADERE_KEY_X,
+        VADERE_KEY_Y,
+    ]
+    name_mapping = {
+        VADERE_KEY_ID: ID_COL,
+        VADERE_KEY_TIME: TIME_COL,
+        VADERE_KEY_X: X_COL,
+        VADERE_KEY_Y: Y_COL,
+    }
+
+    common_error_message = (
+        "The given trajectory file seems to be incorrect or empty. "
+        "It should contain the following columns, which should be "
+        f"uniquely identifiably by: {', '.join(columns_to_keep)}. "
+        f"Columns should be separated by a space character. "
+        f"Comment lines may start with '{VADERE_COMMENT}' and will be ignored. "
+        f"Please check your trajectory file: {trajectory_file}."
+    )
+    try:
+        vadere_cols = list(
+            pd.read_csv(
+                trajectory_file, comment=VADERE_COMMENT, delimiter=" ", nrows=1
+            ).columns
+        )
+        use_vadere_cols = list()
+        non_unique_cols = dict()
+        missing_cols = list()
+        rename_mapping = dict()
+
+        for col in columns_to_keep:
+            matching = [vc for vc in vadere_cols if col in vc]
+            if len(matching) == 1:
+                use_vadere_cols += matching
+                rename_mapping[matching[0]] = name_mapping[col]
+            elif len(matching) > 1:
+                non_unique_cols[col] = matching
+            elif len(matching) == 0:
+                missing_cols += [col]
+
+        if non_unique_cols:
+            raise LoadTrajectoryError(
+                f"{common_error_message} "
+                + ". ".join(
+                    [
+                        "The identifier '{0}' is non-unique. "
+                        "It is contained in the columns: {1}".format(
+                            k, ", ".join(v)
+                        )
+                        for k, v in non_unique_cols.items()
+                    ]
+                )
+                + "."
+            )
+
+        if missing_cols:
+            raise LoadTrajectoryError(
+                f"{common_error_message} "
+                f"Missing columns: {', '.join(missing_cols)}."
+            )
+
+        data = pd.read_csv(
+            trajectory_file,
+            delimiter=" ",
+            usecols=use_vadere_cols,
+            comment="#",
+            dtype={
+                VADERE_KEY_ID: "int64",
+                VADERE_KEY_TIME: "float64",
+                VADERE_KEY_X: "float64",
+                VADERE_KEY_Y: "float64",
+            },
+            encoding="utf-8-sig",
+        )
+
+        data.rename(columns=rename_mapping, inplace=True)
```

Comment:
From the ruff linter: PD002 `inplace=True` should be avoided; it has inconsistent behavior